streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java

```diff
@@ -346,8 +348,14 @@ public void writeAsText(final String filePath,
         return branchChildren;
     }
 
+    
+    @Override 
+    public KStream<K, V> merge(KStream<K, V> stream) {
+        return this.merge(this.builder, (KStream<K, V>) this, stream);
```

Comment:
Nit: remove `this` -- we only use this when necessary.

Also remove the cast -- I guess we need to change `static merge` below for this -- compare my comment there.